packages/project-integrity/src/fiori-project/index.ts

```diff
@@ -71,7 +93,53 @@ export async function initFioriProject(projectRoot: string): Promise<void> {
 export async function checkFioriProjectIntegrity(projectRoot: string): Promise<CheckIntegrityResult> {
     const integrityFilePath = join(projectRoot, fioriIntegrityDataPath);
     const additionalStringContent = await getAdditionalStringContent(projectRoot);
-    return checkProjectIntegrity(integrityFilePath, additionalStringContent);
+    const checkResult = await checkProjectIntegrity(integrityFilePath, additionalStringContent);
+    const integrityData = await readIntegrityData(integrityFilePath);
+    /**
+     * 1. if csn integrity is the same, but file integrity is different, then is compatible changes (e.g empty spaces, new lines or comments). Add them to equal files, update integrity and empty different files,
+     * 2. if csn integrity is different, but file integrity is same, then CDS compiler might have produced different CSN. Remove csnPath from different content and add it to equal content. Update integrity.
+     * 3. if csn integrity is different and file integrity is different, then it is un-compatible changes. Report them.
+     */
+    const csnDiff = checkResult.additionalStringContent?.differentContent.find((content) => content.key === 'csnPath');
+    const fileDiff = checkResult.files.differentFiles.length > 0;
+    if (csnDiff === undefined && fileDiff === true) {
+        // case 1
+        checkResult.files.equalFiles.push(...checkResult.files.differentFiles.map((file) => file.filePath));
+        // also update integrity.json file
+        const diffFileIntegrity = await getFileIntegrity(checkResult.files.differentFiles.map((file) => file.filePath));
+
+        const fileIntegrity = integrityData.fileIntegrity.map((file) => {
+            const diffFile = diffFileIntegrity.find((diff) => diff.filePath === file.filePath);
+            if (diffFile) {
+                return { ...file, hash: diffFile.hash, content: diffFile.content };
+            }
+            return file;
+        });
+        await writeIntegrityData(integrityFilePath, {
+            enabled: integrityData.enabled,
+            fileIntegrity,
+            contentIntegrity: integrityData.contentIntegrity
+        });
+        // empty different files
+        checkResult.files.differentFiles = [];
```

Comment:
Perhaps we only clear `.cds` file changes

Reply:
covered in this patch https://github.com/SAP/open-ux-tools/pull/3096/commits/592c80dbcbcfdda7ece3814cb79723c3dbb65b8e